src/components/ConditionalFilter/API/initialState/useInitialAPIState.tsx

```diff
@@ -0,0 +1,138 @@
+import { useApolloClient } from "@apollo/client";
+import {
+  _GetChannelOperandsDocument,
+  _GetChannelOperandsQuery,
+  _GetChannelOperandsQueryVariables,
+  _SearchAttributeOperandsDocument,
+  _SearchAttributeOperandsQuery,
+  _SearchAttributeOperandsQueryVariables,
+  _SearchCategoriesOperandsDocument,
+  _SearchCategoriesOperandsQuery,
+  _SearchCategoriesOperandsQueryVariables,
+  _SearchCollectionsOperandsDocument,
+  _SearchCollectionsOperandsQuery,
+  _SearchCollectionsOperandsQueryVariables,
+  _SearchProductTypesOperandsDocument,
+  _SearchProductTypesOperandsQuery,
+  _SearchProductTypesOperandsQueryVariables,
+} from "@dashboard/graphql";
+import { useState } from "react";
+
+import { FetchingParams } from "../../ValueProvider/TokenArray/fetchingParams";
+import { InitialStateResponse } from "../InitialStateResponse";
+import { createInitialStateFromData } from "./helpers";
+import { InitialAPIResponse } from "./types";
+
+export interface InitialAPIState {
+  data: InitialStateResponse;
+  loading: boolean;
+  fetchQueries: (params: FetchingParams) => void;
+}
+
+export const useProductInitialAPIState = (): InitialAPIState => {
```

Comment:
question: Shouldn't be `useInitialAPIState`?

Reply:
hmm don't think so, useInitialAPIState suggest type for entire hook, while here he types just the returned value

Reply:
he wants to have initial state for products, perhaps file can be renamed/splited